backend/lib/server_code/api/roomRoutes.js

```diff
@@ -14,24 +19,40 @@ roomRouter.use((req, res, next) => {
   next();
 });
 
-roomRouter.post("/create", (req, res) => {
+roomRouter.post("/create", async (req, res) => {
   const { creatorID } = req.body;
-  //roomPlayers.push(creatorID);
+  const roomID = roomIDCreator();
   roomCreated = true;
-  res.send({ roomID: 1, creatorID, creationDate: res.date });
+  let [socket, socketPort] = [null, null];
+  try {
+    [socket, socketPort] = await createNewSocket();
+    console.log(`socket port got from function:${socketPort}`);
+    socketsCollection[roomID] = [socket, socketPort];
```

Comment:
roomSockets?
anyways, no need to write collection

Reply:
dd1ff27